operator/internal/status/lokistack.go

```diff
@@ -63,18 +61,30 @@ func (e *DegradedError) Error() string {
 	return fmt.Sprintf("cluster degraded: %s", e.Message)
 }
 
-// SetDegradedCondition appends the condition Degraded to the lokistack status conditions.
-func SetDegradedCondition(ctx context.Context, k k8s.Client, req ctrl.Request, msg string, reason lokiv1.LokiStackConditionReason) error {
-	degraded := metav1.Condition{
-		Type:    string(lokiv1.ConditionDegraded),
-		Message: msg,
-		Reason:  string(reason),
+func generateConditions(ctx context.Context, cs *lokiv1.LokiStackComponentStatus, k k8s.Client, req ctrl.Request, stack *lokiv1.LokiStack, degradedErr *DegradedError) ([]metav1.Condition, error) {
+	conditions, err := generateWarnings(ctx, cs, k, req, stack)
+	if err != nil {
+		return nil, err
+	}
+
+	mainCondition, err := generateCondition(ctx, cs, k, req, stack, degradedErr)
+	if err != nil {
+		return nil, err
 	}
 
-	return updateCondition(ctx, k, req, degraded)
+	conditions = append(conditions, mainCondition)
+	return conditions, nil
 }
 
-func generateCondition(ctx context.Context, cs *lokiv1.LokiStackComponentStatus, k k8s.Client, req ctrl.Request, stack *lokiv1.LokiStack) (metav1.Condition, error) {
+func generateCondition(ctx context.Context, cs *lokiv1.LokiStackComponentStatus, k k8s.Client, req ctrl.Request, stack *lokiv1.LokiStack, degradedErr *DegradedError) (metav1.Condition, error) {
```

Comment:
Function has unused parameters:

```suggestion
func generateCondition(ctx context.Context, cs *lokiv1.LokiStackComponentStatus, k k8s.Client, stack *lokiv1.LokiStack, degradedErr *DegradedError) (metav1.Condition, error) {
```